idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/proofs/inspections/CoercionInspectionTest.kt

```diff
@@ -0,0 +1,70 @@
+package arrow.meta.ide.plugins.proofs.inspections
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.testing.IdeTest
+import arrow.meta.ide.testing.Source
+import arrow.meta.ide.testing.env.IdeTestSetUp
+import arrow.meta.ide.testing.env.ideTest
+import arrow.meta.ide.testing.env.types.LightTestSyntax.toKtFile
+import com.intellij.codeInsight.daemon.impl.HighlightInfo
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+
+class CoercionInspectionTest : IdeTestSetUp() {
+
+  override fun setUp() {
+    super.setUp()
+    myFixture.addFileToProject("arrow/prelude.kt", CoercionInspectionTestCode.prelude)
+    myFixture.addFileToProject("consumer/consumer.kt", CoercionInspectionTestCode.twitterHandleDeclaration)
+  }
+
+  @org.junit.Test
+  fun `coercion inspection test`() =
+    ideTest(
+      myFixture = myFixture,
+      ctx = IdeMetaPlugin()
+    ) {
+      listOf<IdeTest<IdeMetaPlugin, Pair<List<HighlightInfo>, String>>>(
+        IdeTest(
+          code = CoercionInspectionTestCode.code1,
+          test = { code: Source, myFixture: CodeInsightTestFixture, _: IdeMetaPlugin ->
+            val file = code.toKtFile(myFixture)
+            val highlightInfos = collectInspections(code, myFixture, listOf(explicitCoercionKtProperty))
+              .filter { it.inspectionToolId == COERCION_EXPLICIT_PROP }
+            val codeFixed = highlightInfos[0].fixFirstInspection(myFixture, file)
+            Pair(highlightInfos, codeFixed)
+          },
+          result = resolvesWhen("CoercionInspectionTest1 for 1 implicit coercion") { pairResult ->
+            pairResult.first.size == 1
+              && pairResult.second == CoercionInspectionTestCode.code1_after_fix
```

Comment:
this can be simplified and reused as a function. Same below

Reply:
how

Reply:
something like: `fun Source.collectAndApplyInspecction( CodeInsightTestFixture, InspectionToo): Pair<List<HighlightInfo>, Source >`, where the returned Source is the code after the Inspection is applied